test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/TxnVerbs.java

```diff
@@ -306,7 +321,15 @@ public static HapiContractCall contractCall(String contract) {
 		return new HapiContractCall(contract);
 	}
 
-	public static HapiContractCall contractCall(String contract, String abi, Object... params) {
+	public static HapiContractCall contractCall(String contract, String functionName, Object... params) {
+		final var abi = getABIFor(FUNCTION, functionName, contract);
+		return new HapiContractCall(abi, contract, params);
+	}
+
+	/** This function provides for the proper execution of specs, which execute contract calls with a function ABI instead of
+	 function name
+	 */
+	public static HapiContractCall contractCallWithFunctionAbi(String contract, String abi, Object... params) {
```

Comment:
Wherever this method is called, a call to `getABIFor` is made prior to calling `contractCallWithFunctionAbi`. I think we can:

- refactor the places where `contractCallWithFunctionAbi` is called to not call `getABIFor` and then this method and call `contractCall` instead
- if the problem is that some function is missing from the contract's json ABI specification, we can manually add it there
- delete `contractCallWithFunctionAbi` method